tests/setup/mockcomponentmanager.js

```diff
@@ -41,6 +41,18 @@ export default class MockComponentManager {
     };
 
     const component = new COMPONENT_REGISTRY[componentType](componentConfig, systemConfig).init(componentConfig);
+
+    if (component.moduleId === undefined || component.moduleId === null) {
+      return component;
+    }
+
+    if (this.core && this.core.globalStorage) {
```

Comment:
would it be a little cleaner to  move the moduleId checks into this second if statement? and are there any tests that don't give components a this.core.globalStorage with an on() method? we might be able to get rid of the check for this.core.globalStorage (maybe we even want it to bug out)

Reply:
actually if you're just trying to match the current logic this makes sense!

Reply:
this could be slightly out of scope, but maybe we want to use the actual ComponentManager in MockComponentManager? At a quick glance it looks like the main differences are adding the 'enzyme-component' css class and setting the handlebars renderer?

Reply:
I'll update this to match your first suggestion, as that matches the current logic in the real ComponentManager. As for using the real ComponentManager, I don't think there's an immediate need but we can always do that if needed